framework/src/userobject/SolutionUserObject.C

```diff
@@ -288,8 +288,18 @@ SolutionUserObject::readExodusII()
   for (const auto & var_name : _elemental_variables)
     _system->add_variable(var_name, CONSTANT, MONOMIAL);
 
-  for (const auto & var_name : _scalar_variables)
-    _system->add_variable(var_name, FIRST, SCALAR);
+  // Handle scalars
+  if (_scalar_variables.size() > 0)
+  {
+    // Iterate over the scalar variables. If we find one matching a nodal variable name, we ignore
+    // it, as this value is a postprocessor outputted to the global variables, not a scalar
+    // variable.
+    for (const std::string & var : _scalar_variables)
+      if (!std::any_of(_nodal_variables.begin(),
```

Comment:
Just a non-action comment: You might also consider `std::find` for comparing all values in a container. It's arguably simpler since it doesn't require a lambda.